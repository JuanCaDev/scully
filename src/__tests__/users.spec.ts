import {readFileSync} from 'fs';
import {join} from 'path';
import {replaceIndexNG} from '../test-config.helper';

describe('JsonPlugin: test', () => {
  it('Check clean blog index by scully', () => {
    const index: string = readFileSync(
      join(__dirname, '../../dist/static/blog/index.html'),
      'UTF8'
    ).toString();
    const cleanIndex = replaceIndexNG(index);
    expect(cleanIndex).toMatchSnapshot();
  });
});
