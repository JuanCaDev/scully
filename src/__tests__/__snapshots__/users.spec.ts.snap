// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonPlugin: test Check clean blog index by scully 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head>
    <meta charset=\\"utf-8\\">
    <title>SampleBlog</title>
    <base href=\\"/\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">
  <link rel=\\"stylesheet\\" href=\\"styles.css\\"><script id=\\"scully-transfer-state\\"></script><style>[]{display:grid;height:100vh;grid-template-rows:60px 1fr 60px}[] > header[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding:0 10px;grid-template-columns:1fr 40px;place-items:center center}footer[]   h3[], header[]   h1[]{margin:0}[] > footer[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding-right:10px;justify-content:right;align-content:center}main[]{background-color:#f5f5f5;color:#040c24;padding:10px}</style><script charset=\\"utf-8\\" src=\\"blog-blog-module-es2015.js\\"></script><script charset=\\"utf-8\\" src=\\"pagenotfound-pagenotfound-module-es2015.js\\"></script><style>[]{display:grid;grid-template-columns:repeat(3,1fr);gap:10px}[]   h1[]:first-child{grid-column:span 3;text-align:center}article[]{background-color:#4169e1;padding:5px;border-radius:5px;color:#f5f5f5}</style><script>window['ScullyIO']='generated';</script></head>

  <body scully-version=\\"0.0.0\\">
    <app-root =\\"\\" > Scully demo blog app! <small =\\"\\">rendering inside scully</small></h1><a =\\"\\" href=\\"/home\\">üè†</a></header><main =\\"\\"><router-outlet =\\"\\"></router-outlet><app-blog-list =\\"\\"><h1 =\\"\\">Overview of blog posts</h1><a =\\"\\" href=\\"/blog/page-1\\"><article =\\"\\"><h3 =\\"\\">My first page</h3><h4 =\\"\\"></h4><p =\\"\\">This is the first demo page in this sample.</p></article></a><a =\\"\\" href=\\"/blog/page-3\\"><article =\\"\\"><h3 =\\"\\">My third page</h3><h4 =\\"\\"></h4><p =\\"\\">At this point, I should write something different in here.</p></article></a><a =\\"\\" href=\\"/blog/page-4\\"><article =\\"\\"><h3 =\\"\\">The Rainbow Cat, A free book out of the Guttenberg library</h3><h4 =\\"\\"></h4><p =\\"\\"></p></article></a><!----></app-blog-list><!----></main><footer =\\"\\"><h3 =\\"\\">Made with ‚ù§Ô∏è <strong =\\"\\">@HeroDevs</strong></h3></footer></app-root>
  <script src=\\"runtime-es2015.js\\" type=\\"module\\"></script><script src=\\"runtime-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script><script src=\\"polyfills-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script><script src=\\"polyfills-es2015.js\\" type=\\"module\\"></script><script src=\\"main-es2015.js\\" type=\\"module\\"></script><script src=\\"main-es5.js\\" nomodule=\\"\\" defer=\\"\\"></script>

</body></html>"
`;
